libvast/src/arrow_table_slice.cpp

```diff
@@ -386,14 +384,15 @@ std::pair<type, std::shared_ptr<arrow::RecordBatch>> transform_columns(
     return result;
   };
   if (transformations.empty())
-    return {schema, batch};
+    return {schema, struct_array};
   auto current = transformations.begin();
   const auto sentinel = transformations.end();
   auto layer = unpacked_layer{
     .fields = {},
-    .arrays = batch->ToStructArray().ValueOrDie()->Flatten().ValueOrDie(),
+    .arrays = struct_array->fields(),
```

Comment:
This needs to be `struct_array->Flatten()`, otherwise we do not merge the null bitmaps correctly.

Reply:
Replaced in https://github.com/tenzir/tenzir/pull/3108/commits/335341e871959d3724ea2fdcf49f5368d2ce5bb2